src/test/java/ResultsServletTest.java

```diff
@@ -0,0 +1,180 @@
+import com.google.sps.servlets.ResultsServlet;
+import com.google.sps.data.TestRecipe;
+import static org.mockito.Mockito.*;
+import org.mockito.junit.MockitoJUnitRunner;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.Assert;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
+import com.google.gson.Gson;
+
+@RunWith(JUnit4.class)
+public class ResultsServletTest {
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private DatastoreService testDatastore;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+
+    testDatastore = DatastoreServiceFactory.getDatastoreService();
+
+    Entity recipeEntity1 = new Entity("Recipe");
+    List<String> searchStrings1 = new ArrayList<String>();
```

Comment:
I think you can simply do:
List<String> searchStrings1 = Arrays.asList("EGG", "FLOUR");

Reply:
fixed~